toolkit/offchain/src/reserve/init.rs

```diff
@@ -296,25 +297,21 @@ pub(crate) async fn find_script_utxo<
 	version_oracle: &VersionOracleData,
 	ctx: &TransactionContext,
 	client: &T,
-) -> Result<Option<UtxoId>, anyhow::Error> {
+) -> Result<Option<OgmiosUtxo>, anyhow::Error> {
 	let validator_address = version_oracle.validator.address(ctx.network).to_bech32(None)?;
 	let validator_utxos = client.query_utxos(&[validator_address]).await?;
 	// Decode datum from utxos and check if it contains script id
-	Ok(validator_utxos
-		.into_iter()
-		.find(|utxo| {
-			utxo.clone()
-				.datum
-				.map(|d| d.bytes)
-				.and_then(|bytes| PlutusData::from_bytes(bytes).ok())
-				.and_then(decode_version_oracle_validator_datum)
-				.filter(|datum| {
-					datum.script_id == script_id
-						&& datum.version_oracle_policy_id == version_oracle.policy.script_hash()
-				})
-				.is_some()
-		})
-		.map(|utxo| utxo.to_domain()))
+	Ok(validator_utxos.into_iter().find(|utxo| {
+		utxo.clone()
+			.datum
+			.map(|d| d.bytes)
+			.and_then(|bytes| PlutusData::from_bytes(bytes).ok())
+			.and_then(decode_version_oracle_validator_datum)
+			.is_some_and(|datum| {
+				datum.script_id == script_id
+					&& datum.version_oracle_policy_id == version_oracle.policy.script_hash()
+			})
+	}))
```

Comment:
I did some error in rebase losing your previous comment, sorry and thanks for fixing .is_some here